src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleObject.cs

```diff
@@ -28,6 +28,19 @@ private static void HandleStartObject(JsonSerializerOptions options, ref Utf8Jso
             }
 
             JsonClassInfo classInfo = state.Current.JsonClassInfo;
+
+            if (classInfo.CreateObject is null && classInfo.ClassType == ClassType.Object)
+            {
+                if (classInfo.Type.IsInterface)
```

Comment:
One item to discuss for future checks like this -- should we do these checks during warm-up (when reading properties via reflection) to not affect run-time perf? The disadvantage of doing that is the given property may not actually ever be deserialized if there is never any json payload for that property.

Reply:
Given this is a "NotSupportedCheck", I think we should do it up-front and not affect run-time perf. The user should know that their model is not currently supported to avoid surprising behavior when the payload changes.

Reply:
>  affect run-time perf

Do you have a sense of how impactful this change is to perf? Should we wait till next week or do you think its minimal?

Reply:
> One item to discuss for future checks like this -- should we do these checks during warm-up (when reading properties via reflection) to not affect run-time perf? The disadvantage of doing that is the given property may not actually ever be deserialized if there is never any json payload for that property.

I thought about that and I initially did these check on property add...but this could block also "serialization" not only deserialization, so I thought to be close to "delegate usage" to have the check "only" if we want to use that, to have error only if we use it, otherwise we could block some unexpected scenario. I think that first check `classInfo.CreateObject is null` will be 99% of use cases false it's handled in a good way by jit  AFAIK and maybe also by cpu preditcor, so it' should be "only"(I know that in context of perf this is offensive :smile:) one extra `test x, y`(not sure but my past test on similar code produce that)